2003-05-30  Guido Guenther  <agx@sigxcpu.org>

	* sysdeps/unix/sysv/linux/mips/pread.c: include <stddef.h>
	* sysdeps/unix/sysv/linux/mips/pread.c: handle __NR_N32_pread64,
	 __NR_O32_pread64 and __NR__N64_pread64.
	* sysdeps/unix/sysv/linux/mips/pread64.c: likewise.
	* sysdeps/unix/sysv/linux/mips/pwrite.c: handle __NR_N32_pwrite64,
	  __NR_O32_pwrite64 and __NR__N64_pwrite64.
	* sysdeps/unix/sysv/linux/mips/pwrite64.c: likewise.

Index: sysdeps/unix/sysv/linux/mips//pread.c
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/unix/sysv/linux/mips/pread.c,v
retrieving revision 1.11
diff -u -p -u -r1.11 pread.c
--- sysdeps/unix/sysv/linux/mips//pread.c	30 May 2003 17:41:46 -0000	1.11
+++ sysdeps/unix/sysv/linux/mips//pread.c	23 Jul 2003 11:49:51 -0000
@@ -33,6 +34,24 @@
 #  error "__NR_pread and __NR_pread64 both defined???"
 # endif
 # define __NR_pread __NR_pread64
+#endif
+#ifdef __NR_O32_pread64         /* Newer kernels renamed but it's the same.  */
+# ifdef __NR_O32_pread
+#  error "__NR_O32_pread and __NR_O32_pread64 both defined???"
+# endif
+# define __NR_O32_pread __NR_O32_pread64
+#endif
+#ifdef __NR_N32_pread64         /* Newer kernels renamed but it's the same.  */
+# ifdef __NR_N32_pread
+#  error "__NR_N32_pread and __NR_N32_pread64 both defined???"
+# endif
+# define __NR_N32_pread __NR_N32_pread64
+#endif
+#ifdef __NR_N64_pread64         /* Newer kernels renamed but it's the same.  */
+# ifdef __NR_N64_pread
+#  error "__NR_N64_pread and __NR_N64_pread64 both defined???"
+# endif
+# define __NR_N64_pread __NR_N64_pread64
 #endif
 
 #if defined __NR_pread || __ASSUME_PREAD_SYSCALL > 0
Index: sysdeps/unix/sysv/linux/mips//pread64.c
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/unix/sysv/linux/mips/pread64.c,v
retrieving revision 1.11
diff -u -p -u -r1.11 pread64.c
--- sysdeps/unix/sysv/linux/mips//pread64.c	30 May 2003 17:41:46 -0000	1.11
+++ sysdeps/unix/sysv/linux/mips//pread64.c	23 Jul 2003 11:49:51 -0000
@@ -33,6 +33,24 @@
 # endif
 # define __NR_pread __NR_pread64
 #endif
+#ifdef __NR_O32_pread64         /* Newer kernels renamed but it's the same.  */
+# ifdef __NR_O32_pread
+#  error "__NR_O32_pread and __NR_O32_pread64 both defined???"
+# endif
+# define __NR_O32_pread __NR_O32_pread64
+#endif
+#ifdef __NR_N32_pread64         /* Newer kernels renamed but it's the same.  */
+# ifdef __NR_N32_pread
+#  error "__NR_N32_pread and __NR_N32_pread64 both defined???"
+# endif
+# define __NR_N32_pread __NR_N32_pread64
+#endif
+#ifdef __NR_N64_pread64         /* Newer kernels renamed but it's the same.  */
+# ifdef __NR_N64_pread
+#  error "__NR_N64_pread and __NR_N64_pread64 both defined???"
+# endif
+# define __NR_N64_pread __NR_N64_pread64
+#endif
 
 #if defined __NR_pread || __ASSUME_PREAD_SYSCALL > 0
 
Index: sysdeps/unix/sysv/linux/mips//pwrite.c
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/unix/sysv/linux/mips/pwrite.c,v
retrieving revision 1.11
diff -u -p -u -r1.11 pwrite.c
--- sysdeps/unix/sysv/linux/mips//pwrite.c	30 May 2003 17:41:46 -0000	1.11
+++ sysdeps/unix/sysv/linux/mips//pwrite.c	23 Jul 2003 11:49:51 -0000
@@ -34,6 +34,24 @@
 # endif
 # define __NR_pwrite __NR_pwrite64
 #endif
+#ifdef __NR_O32_pwrite64         /* Newer kernels renamed but it's the same.  */
+# ifdef __NR_O32_pwrite
+#  error "__NR_O32_pwrite and __NR_O32_pwrite64 both defined???"
+# endif
+# define __NR_O32_pwrite __NR_O32_pwrite64
+#endif
+#ifdef __NR_N32_pwrite64         /* Newer kernels renamed but it's the same.  */
+# ifdef __NR_N32_pwrite
+#  error "__NR_N32_pwrite and __NR_N32_pwrite64 both defined???"
+# endif
+# define __NR_N32_pwrite __NR_N32_pwrite64
+#endif
+#ifdef __NR_N64_pwrite64         /* Newer kernels renamed but it's the same.  */
+# ifdef __NR_N64_pwrite
+#  error "__NR_N64_pwrite and __NR_N64_pwrite64 both defined???"
+# endif
+# define __NR_N64_pwrite __NR_N64_pwrite64
+#endif
 
 #if defined __NR_pwrite || __ASSUME_PWRITE_SYSCALL > 0
 
Index: sysdeps/unix/sysv/linux/mips//pwrite64.c
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/unix/sysv/linux/mips/pwrite64.c,v
retrieving revision 1.12
diff -u -p -u -r1.12 pwrite64.c
--- sysdeps/unix/sysv/linux/mips//pwrite64.c	30 May 2003 17:41:46 -0000	1.12
+++ sysdeps/unix/sysv/linux/mips//pwrite64.c	23 Jul 2003 11:49:51 -0000
@@ -33,6 +33,24 @@
 # endif
 # define __NR_pwrite __NR_pwrite64
 #endif
+#ifdef __NR_O32_pwrite64         /* Newer kernels renamed but it's the same.  */
+# ifdef __NR_O32_pwrite
+#  error "__NR_O32_pwrite and __NR_O32_pwrite64 both defined???"
+# endif
+# define __NR_O32_pwrite __NR_O32_pwrite64
+#endif
+#ifdef __NR_N32_pwrite64         /* Newer kernels renamed but it's the same.  */
+# ifdef __NR_N32_pwrite
+#  error "__NR_N32_pwrite and __NR_N32_pwrite64 both defined???"
+# endif
+# define __NR_N32_pwrite __NR_N32_pwrite64
+#endif
+#ifdef __NR_N64_pwrite64         /* Newer kernels renamed but it's the same.  */
+# ifdef __NR_N64_pwrite
+#  error "__NR_N64_pwrite and __NR_N64_pwrite64 both defined???"
+# endif
+# define __NR_N64_pwrite __NR_N64_pwrite64
+#endif
 
 #if defined __NR_pwrite || __ASSUME_PWRITE_SYSCALL > 0
